sqlmodel/main.py

```diff
@@ -585,9 +585,24 @@ def get_config(name: str) -> Any:
             setattr(new_cls, "__abstract__", True)  # noqa: B010
         return new_cls
 
-    # Override SQLAlchemy, allow both SQLAlchemy and plain Pydantic models
     def __init__(
         cls, classname: str, bases: Tuple[type, ...], dict_: Dict[str, Any], **kw: Any
+    ) -> None:
+        return cls.__do_init__(bases, dict_, kw)
```

Comment:
I think this code is right based on stepping through a debugger. Adding a `classname` argument breaks at runtime.

However, the linter doesn't understand that and is complaining. Any advice on how to teach the linter?

Reply:
This is where typeshed handles it. I suspect we'll need to add a similar one for `__do_init__()` somewhere. Not sure where.

https://github.com/python/typeshed/blob/main/stdlib/builtins.pyi#L196-L197